data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -103,132 +103,100 @@ public TestLocalScan(String format) {
 
   private String sharedTableLocation = null;
   private Table sharedTable = null;
-  private List<Record> file1Records = null;
-  private List<Record> file2Records = null;
-  private List<Record> file3Records = null;
-  private List<Record> file1FirstSnapshotRecords = null;
-  private List<Record> file2FirstSnapshotRecords = null;
-  private List<Record> file3FirstSnapshotRecords = null;
-  private List<Record> file1SecondSnapshotRecords = null;
-  private List<Record> file2SecondSnapshotRecords = null;
-  private List<Record> file3SecondSnapshotRecords = null;
 
+  private final Record genericRecord = GenericRecord.create(SCHEMA);
+
+  private final List<Record> file1FirstRecords =  Lists.newArrayList(
```

Comment:
Looks like the goal is to make these immutable, have you considered using `ImmutableList` too?

Reply:
Good idea. Applied.